### PR TITLE
[gitlab] Accept '#' sign in branches / context URLs

### LINKED ISSUE #5129: Workspace creation erroring out on special character #
### Bug description

Some of my branches reference the issue number of the branch. So, my feature branches always start with a "feature/#". When I try to create a workspace on this branch, it fails because it can't find a branch with the url-encoded value of # which is %23. So it fails. 

### Steps to reproduce

Create a branch in gitlab with the special character # in the middle of the branch name immediately after a /. e.g. feature/#12-abcd
Click the gitpod button in gitlab. 

### Expected behavior

New workspace should be created

### Example repository

_No response_

### Anything else?

I'm encountering this in a v0.10.0 of self-hosted gitlab using gitalb-eks-guide repository.

### PR DESCRIPTION
**Pull request by corneliusludmann**
Allow to open GitLab branches with `#` sign (encoded as `%23`).

Test with: https://clu-workspace-creation-5129.staging.gitpod-dev.com/#https://gitlab.com/gp-test-group/gp-test-project/-/tree/issue-%23123

Fixes #5129

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **corneliusludmann** after the PR has been reviewed.

Associated issue: *#5129*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by gtsiolis — COMMENTED** [review]
I was in the neighborhood and tried this in action! Works well for both GL and GH. ✔️

File: components/server/src/gitlab/gitlab-context-parser.spec.ts
Comment (corneliusludmann) [reply components/server/src/gitlab/gitlab-context-parser.spec.ts:547]: I definitely would love it if we would create proper repos for all these (and the GH) tests under the `gitpod-io` group. For the sake of MVP what do you think of creating an issue for that?

Comment (gtsiolis) [reply components/server/src/gitlab/gitlab-context-parser.spec.ts:547]: > I definitely would love it if we would create proper repos for all these (and the GH) tests under the gitpod-io group. For the sake of MVP what do you think of creating an issue for that?

💯 x 💯

**Review by geropl — APPROVED** [review]
Code LGTM, tested and works: :+1:

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 872cf4bc4684e8653b85ee5f1adcc9fea8e76754</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5362#pullrequestreview-747267380" title="Approved">geropl</a>*

Associated issue: *#5129*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
